### PR TITLE
Clean Up Text Extraction

### PR DESCRIPTION
**Pull request by karanataryn**
Addresses Alex's comments in https://github.com/aryn-ai/sycamore/pull/745 and fixes some other bugs.

### CODE REVIEW

File: lib/sycamore/sycamore/transforms/detr_partitioner.py
Comment (alexaryn) [inline lib/sycamore/sycamore/transforms/detr_partitioner.py:509]: This smells funny.  On the one hand, these two calls involve subclasses of the same class and have the same method name.  On the other hand, the calling conventions and inputs are totally different.  At the very least, the `page_data` thing looks like a hack.  I think the cleaner thing to do here is to have different method names when things should be called differently.

File: lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py
Comment (alexaryn) [inline lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py:39]: nit: put the expensive test (is_empty) last

File: lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py
Comment (alexaryn) [inline lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py:58]: If you just return `self.parse_ocr_output(...)` you can sidestep the unsightly annotation with no loss of type checking.

File: lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py
Comment (alexaryn) [inline lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py:54]: It's unexpected to see this immediately after the function is declared with a very broad type hint.  Python inheritance with mypy is a pain.

File: lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py
Comment (alexaryn) [inline lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py:70]: There's a lot loaded into this line.  I'd probably write it out as a normal if-then-else.

File: lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py
Comment (alexaryn) [inline lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py:68]: It looks like pdf_to_image_files() doesn't accept an IOBase.  Should we assert that filename is an instance of str?

File: lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py
Comment (alexaryn) [inline lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py:76]: A little cleaner:
```
for image in generator:
    if isinstance(image, Path):
        image = Image.open(image).convert("RGB")
    ...
```

File: lib/sycamore/sycamore/transforms/text_extraction/pdf_miner.py
Comment (alexaryn) [inline lib/sycamore/sycamore/transforms/text_extraction/pdf_miner.py:28]: Do we need to store these two in `self`?

**Review by alexaryn — COMMENTED** [review]
Overall looks good, but there are a couple of things that could use changing.

File: lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py
Comment (karanataryn) [reply lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py:54]: Agreed :/

File: lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py
Comment (karanataryn) [reply lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py:76]: Could be more than path, so adjusted it for that.

File: lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py
Comment (karanataryn) [reply lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py:70]: Done.

File: lib/sycamore/sycamore/transforms/text_extraction/pdf_miner.py
Comment (karanataryn) [reply lib/sycamore/sycamore/transforms/text_extraction/pdf_miner.py:28]: removed.

File: lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py
Comment (karanataryn) [reply lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py:49]: Doesn't the cast deal with this already? We potentially get a wider range of acceptable entries this way.

File: lib/sycamore/sycamore/transforms/detr_partitioner.py
Comment (karanataryn) [reply lib/sycamore/sycamore/transforms/detr_partitioner.py]: Changed.

File: lib/sycamore/sycamore/transforms/text_extraction/pdf_miner.py
Comment (alexaryn) [inline lib/sycamore/sycamore/transforms/text_extraction/pdf_miner.py:76]: So, I guess `page_data` wasn't needed after all?

**Review by alexaryn — APPROVED** [review]
Looks OK.  Some changes I think are needed before landing.  Approving now so as not to block you.

File: lib/sycamore/sycamore/transforms/text_extraction/pdf_miner.py
Comment (karanataryn) [reply lib/sycamore/sycamore/transforms/text_extraction/pdf_miner.py:76]: Not exactly. The functionality to support it was moved into a `TextExtractor` function.

File: lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py
Comment (karanataryn) [reply lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py:49]: Changed.

File: lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py
Comment (karanataryn) [reply lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py:185]: Done.

File: lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py
Comment (karanataryn) [reply lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py:183]: Good catch, math vs cs always a pain.

**Pull request merged**
